static/js/components/tiles/map_tile.tsx

```diff
@@ -554,108 +574,151 @@ export function draw(
+  const allDataValues: { [placeDcid: string]: { [variable: string]: number } } =
+    {};
+  const allMetadataValues: {
+    [placeDcid: string]: { [variable: string]: DataPointMetadata };
+  } = {};
+  const units: { [variable: string]: string } = {};
+  // mapping of variable to values for computing color scale
+  const allValuesByVariable: { [variable: string]: number[] } = {};
+  const variableNames: { [variable: string]: string } = {};
+  // lists of geojsons used for computing projections and borders
   const projectionGeoJsons = [];
   const bordersToDraw = [];
-  for (const placeData of chartData.placeData) {
+
+  for (const layer of chartData.layerData) {
+    // Build mapping of all values and metadata for a place, to show in tooltip
+    for (const place in layer.dataValues) {
+      if (!Object.keys(allDataValues).includes(place)) {
+        allDataValues[place] = {};
+      }
+      allDataValues[place][layer.variable.statVar] = layer.dataValues[place];
+
+      if (!Object.keys(allMetadataValues).includes(place)) {
+        allMetadataValues[place] = {};
+      }
+      allMetadataValues[place][layer.variable.statVar] = layer.metadata[place];
+    }
+    // Build variable -> values mapping for color scale calculations
+    // Assumes each place and variable combination will only ever have 1 unique
+    // value.
+    allValuesByVariable[layer.variable.statVar] = Object.values(
+      layer.dataValues
+    );
+    variableNames[layer.variable.statVar] = layer.variable.name;
+    units[layer.variable.statVar] = layer.unit;
+
     // Use border data to calculate projection if using borders.
     // This prevents borders from being cutoff when enclosed places don't
     // provide wall to wall coverage.
     const shouldUseBorderData =
-      placeData.enclosedPlaceType &&
-      shouldShowBorder(placeData.enclosedPlaceType) &&
-      !_.isEmpty(placeData.borderGeoJson);
+      layer.enclosedPlaceType &&
+      shouldShowBorder(layer.enclosedPlaceType) &&
+      !_.isEmpty(layer.borderGeoJson);
     projectionGeoJsons.push(
-      shouldUseBorderData ? placeData.borderGeoJson : placeData.geoJson
+      shouldUseBorderData ? layer.borderGeoJson : layer.geoJson
     );
     if (shouldUseBorderData) {
-      bordersToDraw.push(placeData.borderGeoJson);
+      bordersToDraw.push(layer.borderGeoJson);
     }
   }
 
+  // Calculate color scales for each variable and add legends
+  let totalLegendWidth = 0;
+  const legendData: {
+    colorScale: d3.ScaleLinear<number, number, never>;
+    unit: string;
+    label?: string;
+  }[] = [];
+  for (const variable in allValuesByVariable) {
+    // calculate color scale based on max/min values across all layers
+    const dataValues = allValuesByVariable[variable];
+    const colorScale = getColorScale(
+      variable,
+      d3.min(dataValues),
+      d3.mean(dataValues),
+      d3.max(dataValues),
+      undefined,
+      undefined,
+      props.colors
```

Comment:
Good point. Updated so that props.colors is only used if we have one layer (maintaining current behavior of the map tile and web component). Extending custom colors to multiple layers will come in a future PR (after solidifying the spec for how users should specify those colors).